osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/EventConverter.java

```diff
@@ -47,6 +47,11 @@
             event.setTimestamp(timestamp);
             event.setEventCounter(source.getEventCounter());
             event.setEventLogCategory(EventLogCategory.fromValue(source.getEventLogCategory().name()));
+            event.setEventDuration(source.getDuration());
+            if (source.getStartTime() != null) {
+                event.setStartTime(DatatypeFactory.newInstance().newXMLGregorianCalendar(
```

Comment:
if the source would already be an XmlGregorianCalender field you would not need this conversion

Reply:
The conversion is necessary, also see the existing timestamp.